cli/cmd/deploy/deploy.go

```diff
@@ -200,74 +200,181 @@ func DeployCmd(cfg *config.Config) *cobra.Command {
+			case <-pollCtx.Done():
+				return nil, pollCtx.Err()
+			case <-time.After(pollInterval):
+				// Ready to check again.
+			}
+
+			// Poll for access to the Github URL
+			pollRes, err := c.GetGithubRepoStatus(ctx, &adminv1.GetGithubRepoStatusRequest{
+				GithubUrl: githubURL,
+			})
+			if err != nil {
+				return nil, err
+			}
+
+			if pollRes.HasAccess {
+				// Success
+				return pollRes, nil
+			}
+
+			// Sleep and poll again
 		}
+	}
 
-		return org.Create(adminClient, name, "")
+	return res, nil
+}
+
+func variablesFlow(projectPath string) (map[string]string, error) {
+	connectors, err := rillv1beta.ExtractConnectors(projectPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract connectors %w", err)
 	}
-	return resp, err
+
+	vars := make(map[string]string)
+	for _, c := range connectors {
+		connectorVariables := c.Spec.ConnectorVariables
+		if len(connectorVariables) != 0 {
+			fmt.Printf("\nConnector %s requires credentials\n\n", c.Type)
+		}
+		if c.Spec.Help != "" {
+			fmt.Println(c.Spec.Help)
+		}
+		for _, prop := range connectorVariables {
+			question := &survey.Question{}
+			msg := fmt.Sprintf("connector.%s.%s", c.Name, prop.Key)
+			if prop.Help != "" {
+				msg = fmt.Sprintf(msg+" (%s)", prop.Help)
+			}
+
+			if prop.Secret {
+				question.Prompt = &survey.Password{Message: msg}
+			} else {
+				question.Prompt = &survey.Input{Message: msg, Default: prop.Default}
+			}
+
+			if prop.TransformFunc != nil {
+				question.Transform = prop.TransformFunc
+			}
+
+			if prop.ValidateFunc != nil {
+				question.Validate = prop.ValidateFunc
+			}
+
+			answer := ""
+			if err := survey.Ask([]*survey.Question{question}, &answer); err != nil {
+				return nil, fmt.Errorf("variables prompt failed with error %w", err)
+			}
+
+			if answer != "" {
+				vars[prop.Key] = answer
+			}
+		}
+	}
+	return vars, nil
 }
 
-func createProject(ctx context.Context, adminClient *client.Client, req *adminv1.CreateProjectRequest) (*adminv1.CreateProjectResponse, error) {
-	// Create the project (automatically deploys prod branch)
-	res, err := adminClient.CreateProject(ctx, req)
-	if err != nil && violatesUniqueConstraint(err) {
-		// project name already exists, prompt for project name and create project with new name again
-		name, err := projectNamePrompt(ctx, adminClient, req.OrganizationName)
+func createOrgFlow(ctx context.Context, cfg *config.Config, client *adminclient.Client, defaultName string) error {
+	res, err := client.CreateOrganization(ctx, &adminv1.CreateOrganizationRequest{
+		Name: defaultName,
+	})
+	if err != nil {
+		if !isNameExistsErr(err) {
+			return err
+		}
+
+		name, err := orgNamePrompt(ctx, client)
 		if err != nil {
-			return nil, err
+			return err
 		}
 
-		req.Name = name
-		return adminClient.CreateProject(ctx, req)
+		res, err = client.CreateOrganization(ctx, &adminv1.CreateOrganizationRequest{
+			Name: name,
+		})
+		if err != nil {
+			return err
+		}
 	}
-	return res, err
+
```

Comment:
This can be moved into a common create org helper function in `cli/cmd/org/create.go`. Makes it easy to make any further changes in org creation flow.